org.eclipse.winery.topologymodeler.ui/src/app/node/properties-content/properties-content.component.ts

```diff
@@ -43,21 +43,19 @@ export class PropertiesContentComponent implements OnInit, OnChanges, OnDestroy
      * Angular lifecycle event.
      */
     ngOnChanges(changes: SimpleChanges) {
-        setTimeout(() => {
-            if (this.currentNodeData.currentNodePart === 'PROPERTIES') {
-                if (changes.currentNodeData.currentValue.nodeTemplate.properties) {
-                    try {
-                        const currentProperties = changes.currentNodeData.currentValue.nodeTemplate.properties;
-                        if (this.currentNodeData.propertyDefinitionType === 'KV') {
-                            this.nodeProperties = currentProperties.kvproperties;
-                        } else if (this.currentNodeData.propertyDefinitionType === 'XML') {
-                            this.nodeProperties = currentProperties.any;
-                        }
-                    } catch (e) {
+        if (this.currentNodeData.currentNodePart === 'PROPERTIES') {
+            if (changes.currentNodeData.currentValue.nodeTemplate.properties) {
+                try {
+                    const currentProperties = changes.currentNodeData.currentValue.nodeTemplate.properties;
+                    if (this.currentNodeData.propertyDefinitionType === 'KV') {
```

Comment:
Could you use a string enum for this?
See https://blog.mariusschulz.com/2017/10/27/typescript-2-4-string-enums for a how to.